qiskit/synthesis/evolution/pauli_network.py

```diff
@@ -0,0 +1,80 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Circuit synthesis for pauli evolution gates."""
+
+from __future__ import annotations
+
+from qiskit.circuit import QuantumCircuit
+
+from qiskit._accelerate.synthesis.evolution import (
+    pauli_network_synthesis as pauli_network_synthesis_inner,
+)
+
+
+def synth_pauli_network_rustiq(
+    num_qubits: int,
+    pauli_network: list,
+    optimize_count: bool,
+    preserve_order: bool,
+    upto_clifford: bool,
+    upto_phase: bool,
+    resynth_clifford_method: int,
+) -> QuantumCircuit:
+    """
+    Calls Rustiq's pauli network synthesis algorithm.
+
+    The algorithm is described in [1]. The source code (in Rust) is available at
+    https://github.com/smartiel/rustiq-core.
```

Comment:
should the user install and compile rustiq-core for this code ?

Reply:
No, the rustiq-core (version 0.0.8) is available [crates.io](https://crates.io/crates/rustiq-core), and it's "pulled in" during Qiskit's Rust compilation (as all the other Rust crates Qiskit internally uses). There is nothing specific a user needs to install.